security/leaks.js

```diff
@@ -0,0 +1,140 @@
+window.addEventListener('load', async () => {
+    const sections = {};
+    ['added', 'removed', 'changed'].forEach((s) => {
+        sections[s] = document.getElementById(s);
+    });
+
+    function addBullet (html, section) {
+        const li = document.createElement('li');
+        li.innerHTML = html;
+        sections[section].appendChild(li);
+    }
+
+    function collectProps () {
+        const objCache = new Set();
+        // skip these
+        objCache.add(document);
+        objCache.add(window.self);
+        objCache.add(document.activeElement);
+        objCache.add(event?.currentTarget);
+        const recursionAllowed = new Set();
+        recursionAllowed.add('navigator');
+
+        function extractObjectProps (obj, bc = []) {
+            const props = {};
+            const proplist = [];
+            for (const prop in obj) {
+                proplist.push(prop);
+            }
+            proplist.sort().forEach((prop) => {
+                if (bc.length === 0 && prop === 'collectProps') {
```

Comment:
if you are checking for global that this page adds, this global is called `collectedProps`, also please check for `results` (that's where we want to export the results from the test for automatic tests to pick them up)